app/Controller/FeedsController.php

```diff
@@ -190,262 +197,268 @@ public function importFeeds()
+                        $feed['Feed']['orgc_id'] = '0';
+                    }
+                }
+                if ($feed['Feed']['source_format'] == 'freetext') {
+                    if ($feed['Feed']['fixed_event'] == 1) {
+                        if (!empty($feed['Feed']['target_event']) && is_numeric($feed['Feed']['target_event'])) {
+                            $feed['Feed']['event_id'] = $feed['Feed']['target_event'];
+                        }
                     }
-                    $this->Flash->success($message);
-                    $this->redirect(array('controller' => 'feeds', 'action' => 'index'));
+                }
+                if (!isset($feed['Feed']['settings'])) {
+                    $feed['Feed']['settings'] = array();
                 } else {
-                    $message = __('Feed could not be added. Reason: %s', json_encode($this->Feed->validationErrors));
-                    if ($this->_isRest()) {
-                        return $this->RestResponse->saveFailResponse('Feeds', 'add', false, $message, $this->response->type());
+                    if (!empty($feed['Feed']['settings']['common']['excluderegex']) && !$this->__checkRegex($feed['Feed']['settings']['common']['excluderegex'])) {
+                        $regexErrorMessage = __('Invalid exclude regex. Make sure it\'s a delimited PCRE regex pattern.');
+                        if (!$this->_isRest()) {
+                            $this->Flash->error($regexErrorMessage);
+                            return true;
+                        } else {
+                            return new CakeResponse(array(
+                                'body' => json_encode(array('saved' => false, 'errors' => $regexErrorMessage)),
+                                'status' => 200,
+                                'type' => 'json'
+                            ));
+                        }
                     }
-                    $this->Flash->error($message);
-                    $this->request->data['Feed']['settings'] = json_decode($this->request->data['Feed']['settings'], true);
                 }
+                if (isset($feed['Feed']['settings']['delimiter']) && empty($feed['Feed']['settings']['delimiter'])) {
+                    $feed['Feed']['settings']['delimiter'] = ',';
+                }
+                if (empty($feed['Feed']['target_event'])) {
+                    $feed['Feed']['target_event'] = 0;
+                }
+                if (empty($feed['Feed']['lookup_visible'])) {
+                    $feed['Feed']['lookup_visible'] = 0;
+                }
+                if (empty($feed['Feed']['input_source'])) {
+                    $feed['Feed']['input_source'] = 'network';
+                } else {
+                    $feed['Feed']['input_source'] = strtolower($feed['Feed']['input_source']);
+                }
+                if (!in_array($feed['Feed']['input_source'], array('network', 'local'))) {
```

Comment:
Should we also enforce `Security.disable_local_feed_access` here? (As well as for other endpoints)

Reply:
looked into it and it's already checked here: https://github.com/MISP/MISP/blame/2.4/app/Model/Feed.php#L83